synaptogram/views.py

```diff
@@ -251,6 +252,60 @@ def sgram_from_ndviz(request):
     return HttpResponseRedirect(reverse('synaptogram:cutout', args=(coll,exp)) + params)
     #redirect('synaptogram:cutout', coll=coll,exp=exp) 
 
+@login_required
+def channel_detail(request, coll, exp, channel):
+    add_url = '/'.join(('collection', coll, 'experiment', exp, 'channel', channel))
+    url, headers = get_boss_request(request, add_url)
+    resp = get_resp_from_boss(request, url, headers)
+    request.session['next'] = '/exp_list/' + coll
+    if resp == 'authentication failure':
+        return redirect('/openid/openid/KeyCloak')
+    ch_props = resp
+    
+    add_url = 'permissions/?' + '&'.join(('collection=' + coll, 'experiment=' + exp, 'channel=' + channel))
```

Comment:
I like the python string formatter better than string operations. E.g. `'&'.join('collection={}'.format(coll), 'experiment={}'.format(exp), 'channel={}'.format(channel))`. That way if you happen to have a numeric channel the proper conversion will be called to convert the number to a string and the return the full `channel=5`, rather than crashing with an error. See https://pyformat.info/ for more.

Not to say you should change all of this, but worth considering for the future.

Reply:
Thanks! That's a great link.